packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -1425,7 +1426,7 @@ abstract class State<T extends StatefulWidget> with Diagnosticable {
   ///
   ///  * [StatefulWidget], which contains the discussion on performance considerations.
   @protected
-  Widget build(BuildContext context);
+  Widget? build(BuildContext context);
```

Comment:
this should be non-nullable

Reply:
It would conflict with `AutomaticKeepAliveClientMixin.build` :

```dart
  @mustCallSuper
  @override
  Widget? build(BuildContext context) {
    if (wantKeepAlive && _keepAliveHandle == null)
      _ensureKeepAlive();
    return null;
  }
```

Reply:
we need to change `AutomaticKeepAliveClientMixin.build`, something like this should do (untested):

```diff
diff --git a/packages/flutter/lib/src/widgets/automatic_keep_alive.dart b/packages/flutter/lib/src/widgets/automatic_keep_alive.dart
index b2ae057b3f..9b6e11d5f9 100644
--- a/packages/flutter/lib/src/widgets/automatic_keep_alive.dart
+++ b/packages/flutter/lib/src/widgets/automatic_keep_alive.dart
@@ -330,8 +330,7 @@ class KeepAliveHandle extends ChangeNotifier {
 /// with [State] subclasses.
 ///
 /// Subclasses must implement [wantKeepAlive], and their [build] methods must
-/// call `super.build` (the return value will always return null, and should be
-/// ignored).
+/// call `super.build` (though the return value should be ignored).
 ///
 /// Then, whenever [wantKeepAlive]'s value changes (or might change), the
 /// subclass should call [updateKeepAlive].
@@ -397,6 +396,17 @@ mixin AutomaticKeepAliveClientMixin<T extends StatefulWidget> on State<T> {
   Widget build(BuildContext context) {
     if (wantKeepAlive && _keepAliveHandle == null)
       _ensureKeepAlive();
-    return null;
+    return const _NullWidget();
+  }
+}
+
+class _NullWidget extends StatelessWidget {
+  const _NullWidget();
+  @override
+  Widget build(BuildContext context) {
+    throw FlutterError(
+      'Widgets that mix AutomaticKeepAliveClientMixin into their State must '
+      'call super.build() but must ignore the return value of the superclass.'
+    );
   }   
 }  
```

Reply:
Done